python/covid_graphs/covid_graphs/predictions.py

```diff
@@ -196,5 +204,77 @@ def predictions_for_event(self, prediction_event: PredictionEvent) -> List[Count
     def predictions_for_country(self, country: str) -> List[CountryPrediction]:
         return [p for p in self._prediction_database if p.country == country]
 
+    def select_predictions(
+        self, country: str, last_data_dates: List[datetime.date]
+    ) -> List[CountryPrediction]:
+        return [
+            p
+            for p in self._prediction_database
+            if p.country == country and p.prediction_event.last_data_date in last_data_dates
+        ]
+
+
+def load_prediction_db(prediction_dir: Path) -> PredictionDb:
+    country_predictions = _prediction_database[:]
+
+    # TODO: Load all predictions from prediction dir. Make the folder existence mandatory.
+    if not prediction_dir.is_dir():
+        logging.warning(f"Could not load predictions from {prediction_dir}.")
+
+    for country_atg_params_path in prediction_dir.glob("*.atg"):
+        country_atg_parameters = CountryAtgParameters()
+        text_format.Parse(country_atg_params_path.read_text(), country_atg_parameters)
+
+        fitted_formulas = [
+            formula.create_formula_from_proto(atg_parameters)
+            for atg_parameters in country_atg_parameters.parameters
+        ]
+        # TODO(miskosz): The decision on which predictions to display should be made by the caller.
+        start_idx = _calculate_earliest_displayable_idx(fitted_formulas, MAX_PEAK_DISTANCE)
+        fitted_predictions = _create_predictions_from_formulas(
+            fitted_formulas[start_idx:], country_atg_parameters.short_country_name
+        )
+        country_predictions.extend(fitted_predictions)
+
+    return PredictionDb(country_predictions=country_predictions)
+
+
+# TODO(miskosz): Move to the caller.
```

Comment:
Can you clarify what you mean by moving to the caller?

Reply:
I'll rephrase. I don't think that PredictionDb should filter data that is available, it should serve everything and the user should choose what they want. Choosing which traces to display can be responsibility of the dashboard; or we can move it to prediction generation (so that uncertain predictions are not even exported). My preference would be the former.

Reply:
Right, I agree, but please expand the TODO a bit.